compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -890,3 +890,136 @@ impl SirTypes {
+        let data: Vec<&str> = s.split(':').collect();
+        debug_assert!(data.len() == 3);
+        let sym = data[1].to_owned();
+        let bb_idx = data[2].parse::<u32>().unwrap();
+        (sym, bb_idx)
+    }
+
+    pub fn add_yk_label_section(exe_path: &Path) {
+        let labels = extract_dwarf_labels(exe_path).unwrap();
+        let mut tempf = tempfile::NamedTempFile::new().unwrap();
+        bincode::serialize_into(&mut tempf, &labels).unwrap();
+        add_section(exe_path, tempf.path());
+    }
+
+    fn add_section(exe_path: &Path, sec_data_path: &Path) {
+        let mut out_path = PathBuf::from(exe_path);
+        out_path.set_extension("with_labels");
+        Command::new("objcopy")
+            .args(&[
+                "--add-section",
+                &format!("{}={}", ykpack::YKLABELS_SECTION, sec_data_path.to_str().unwrap()),
+                "--set-section-flags",
+                &format!("{}=contents,alloc,readonly", ykpack::YKLABELS_SECTION),
+                exe_path.to_str().unwrap(),
+                out_path.to_str().unwrap(),
+            ])
+            .output()
+            .expect("failed to insert labels section");
+        std::fs::rename(out_path, exe_path).unwrap();
+    }
+
+    fn extract_dwarf_labels(exe_filename: &Path) -> Result<Vec<ykpack::SirLabel>, gimli::Error> {
+        let file = fs::File::open(exe_filename).unwrap();
+        let mmap = unsafe { memmap::Mmap::map(&file).unwrap() };
+        let object = object::File::parse(&*mmap).unwrap();
+        let endian = if object.is_little_endian() {
+            gimli::RunTimeEndian::Little
+        } else {
+            gimli::RunTimeEndian::Big
+        };
+        let mut labels = Vec::new();
+
+        let loader = |id: gimli::SectionId| -> Result<&[u8], gimli::Error> {
+            Ok(object
+                .section_by_name(id.name())
+                .map(|sec| sec.data().expect("failed to decompress section"))
+                .unwrap_or(&[] as &[u8]))
+        };
+        let sup_loader = |_| Ok(&[] as &[u8]);
+        let dwarf_cow = gimli::Dwarf::load(&loader, &sup_loader)?;
+        let borrow_section: &dyn for<'a> Fn(
+            &&'a [u8],
+        )
+            -> gimli::EndianSlice<'a, gimli::RunTimeEndian> =
+            &|section| gimli::EndianSlice::new(section, endian);
+        let dwarf = dwarf_cow.borrow(&borrow_section);
+        let mut iter = dwarf.units();
+        let mut subaddr = None;
+        while let Some(header) = iter.next()? {
+            let unit = dwarf.unit(header)?;
+            let mut entries = unit.entries();
+            while let Some((_, entry)) = entries.next_dfs()? {
+                if entry.tag() == gimli::DW_TAG_subprogram {
+                    if let Some(_name) = entry.attr_value(gimli::DW_AT_linkage_name)? {
+                        if let Some(lowpc) = entry.attr_value(gimli::DW_AT_low_pc)? {
+                            let addr = match lowpc {
```

Comment:
This might be easier as:

```rust
if let gimli::AttributeValue::Addr(v) = v {
    addr = Some(v);
} else {
    panic!("...");
}
```

Reply:
I must admit, I copy-pasted that whole function blindly. Will try it.